blacktie/tx/src/main/cpp/HttpTxManager.cxx

```diff
@@ -379,7 +376,7 @@ bool HttpTxManager::handle_request(
 					LOG4CXX_DEBUG(httptxlogger, "Return 200 OK with body: " <<
 						status << " XA status: " << res);
 					code = 200;
-					_branches.erase(branch->get_name());
+					_branches.erase(wid);
```

Comment:
I propose to add the removing function in the HttpTxManager and when the XAResourceManager deletes the branch, it notifies the HttpTxManager to remove it too.
```
void HttpTxManager::remove_branch(const char * name) {
    _branches.erase(name); // does it need to add the lock ?
}
```
https://github.com/jbosstm/narayana/blob/master/blacktie/tx/src/main/cpp/XAResourceManager.cxx#L189
```
bool XAResourceManager::recoverRTS(XID& bid, const char* rc) {
...
    atmibroker::tx::TxManager::get_instance()->remove_branch(ra->get_name());
    delete ra;
}
```
https://github.com/jbosstm/narayana/blob/master/blacktie/tx/src/main/cpp/XAResourceManager.cxx#L328
```
void XAResourceManager::set_complete(XAWrapper* resource) {
...
    LOG4CXX_TRACE(xarmlogger, (char*) "removing branch");
    atmibroker::tx::TxManager::get_instance()->remove_branch(ra->get_name());
    delete ra;
}
```